djhtml/middleware.py

```diff
@@ -0,0 +1,43 @@
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+
+from djhtml.modes import DjHTML
+
+try:
+    TABWIDTH = settings.TABWIDTH
+except AttributeError:
+    TABWIDTH = 4
+
+
+class DjHTMLMiddleware:
+    """
+    Django middleware class to indent HTML responses.
+
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+        if (
+            "Content-Type" in response
+            and response["Content-Type"] == "text/html; charset=utf-8"
+        ):
+            if "Content-Length" in response:
+                raise ImproperlyConfigured(
+                    "Please load DjHTMLMiddleware _after_ CommonMiddleware (otherwise"
+                    ' the "Content-Length" header will be incorrect)'
+                )
+
+            response.content = "\n".join(
```

Comment:
The first point has already been addressed, see https://github.com/rtts/djhtml/blob/9f37df8efe6fb9ea20f2d6b49d6527cffd26104b/djhtml/middleware.py#L23

        if (
            "Content-Type" in response
            and response["Content-Type"] == "text/html; charset=utf-8"
        ):

The second and third points are very valid, so I will wrap the whole thing in a `try... except Exception` statement and return the original content instead.

The fourth point is moot, because who generates template tags in a response? And even if they're incorrectly indented, the HTML would still be valid.

Reply:
Never mind, I have removed this problematic middleware entirely.